code/ModelAdmin.php

```diff
@@ -183,6 +185,55 @@ public function Link($action = null)
         return parent::Link($action);
     }
 
+    /**
+     * Get the link for the tab of a DataObject subclass managed by this ModelAdmin.
+     *
+     * Note that if the modelClass is managed on multiple tabs and you want a link for a
+     * specific tab you should use {@link getLinkForModelTab} instead.
+     *
+     * @throws InvalidArgumentException if $modelClass is not managed by this ModelAdmin.
+     */
+    public function getLinkForModelClass(string $modelClass): string
+    {
+        if (!$this->isManagedModel($modelClass)) {
+            throw new InvalidArgumentException("$modelClass isn't managed by this ModelAdmin.");
+        }
+        return $this->getLinkForModelTab($this->getModelTabForModelClass($modelClass));
+    }
+
+    /**
+     * Get the link for a specific tab on this ModelAdmin.
+     *
+     * If you do not have multiple tabs for a given class you should use {@link getLinkForModelClass}
+     * instead.
+     *
+     * @throws InvalidArgumentException if $modelTab is not a tab on this ModelAdmin.
+     */
+    public function getLinkForModelTab(string $modelTab): string
+    {
+        // Don't use isManagedModel here because a subclass of a managed model may not have its own tab.
+        if (!array_key_exists($modelTab, $this->getManagedModels())) {
+            throw new InvalidArgumentException("$modelTab isn't a tab on this ModelAdmin.");
+        }
+        return $this->Link($this->sanitiseClassName($modelTab));
+    }
+
+    /**
+     * Get the link for editing an object inside this ModelAdmin.
+     */
+    public function getEditLinkForDataObject(DataObject $obj): string
+    {
+        $link = static::join_links(
+            $this->getLinkForModelClass($obj->ClassName),
+            'EditForm/field/',
+            $this->sanitiseClassName($this->getModelTabForModelClass($obj->ClassName)),
```

Comment:
`getModelTabForModelClass()` can return null.

I think it's worth doing something like that at the top of the method:

```php
$modelTab = $this->getModelTabForModelClass($obj->ClassName);
if (is_null($modelTabl) {
    throw new InvalidArgumentException(sprintf('There is no ModelTab for ModelClass %s', $obj->ClassName));
}
```

Reply:
An exception like this is already thrown by `getLinkForModelClass()` which is called from here, but it makes sense to add an explicit exception here too.

Reply:
Added the exception